tests/rules/test_script_failed.py

```diff
@@ -0,0 +1,168 @@
+        assert rule.name in repr(rule)
+        assert "ScriptFailed" in repr(rule)
+
+    async def test_call(self):
+        index = 1
+
+        watcher_config_dict = yaml.safe_load(
+            f"""
+            disabled_sal_components: []
+            auto_acknowledge_delay: 3600
+            auto_unacknowledge_delay: 3600
+            rules:
+            - classname: ScriptFailed
+              configs:
+              - index: {index}
+            escalation: []
+            """
+        )
+        watcher_config = types.SimpleNamespace(**watcher_config_dict)
+
+        async with salobj.Controller(name="ScriptQueue", index=index) as script_queue:
+            async with watcher.Model(
+                domain=script_queue.domain, config=watcher_config
+            ) as model:
+                model.enable()
+
+                assert len(model.rules) == 1
+                rule_name = f"ScriptFailed.ScriptQueue:{index}"
+                rule = model.rules[rule_name]
+                rule.alarm.init_severity_queue()
+
+                for (
+                    queue_enabled,
+                    queue_running,
+                    script_sal_index,
+                    process_state,
+                    script_state,
+                    expected_severity_1,
+                    expected_severity_2,
+                ) in [
+                    (
+                        True,
+                        True,
+                        1000,
+                        ScriptProcessState.RUNNING,
+                        ScriptState.RUNNING,
+                        AlarmSeverity.NONE,
+                        AlarmSeverity.NONE,
+                    ),
+                    (
+                        True,
+                        True,
+                        1001,
+                        ScriptProcessState.DONE,
+                        ScriptState.DONE,
+                        AlarmSeverity.NONE,
+                        AlarmSeverity.NONE,
+                    ),
+                    (
+                        True,
+                        False,
+                        1002,
+                        ScriptProcessState.DONE,
+                        ScriptState.FAILED,
+                        AlarmSeverity.NONE,
+                        AlarmSeverity.WARNING,
+                    ),
+                ]:
+
+                    await script_queue.evt_queue.set_write(
+                        enabled=queue_enabled,
+                        running=queue_running,
+                        currentSalIndex=script_sal_index,
+                    )
+
+                    await script_queue.evt_script.set_write(
+                        scriptSalIndex=script_sal_index,
+                        processState=process_state,
+                        scriptState=script_state,
+                    )
+
+                    # It will publish 2 alarms one for each event published
+                    # by the ScriptQueue.
```

Comment:
I suggest clarifying this comment. I believe this will publish 2 alarm states for each set of test parameters: one for the queue ScriptQueue event, the next for the script ScriptQueue event. (I read your current text to mean we get 2 alarm states per single ScriptQueue event, which surely is not right).